server/command/service.go

```diff
@@ -30,52 +30,138 @@ type service struct {
 
 var _ Service = (*service)(nil)
 
-func MakeService(mm *pluginapi.Client, configService config.Service, proxy proxy.Service, httpOut httpout.Service) (Service, error) {
-	s := &service{
-		mm:      mm,
-		conf:    configService,
-		proxy:   proxy,
-		httpOut: httpOut,
-	}
-	conf := configService.GetConfig()
-	subCommands := s.getSubCommands()
-	var subTrigger []string
-	for t, c := range subCommands {
-		if c.debug && !conf.DeveloperMode {
-			continue
-		}
+type params struct {
+	pluginContext *plugin.Context
+	commandArgs   *model.CommandArgs
+	current       []string
+}
+
+type commandHandler struct {
+	f            func(*params) (*model.CommandResponse, error)
+	devOnly      bool
+	autoComplete *model.AutocompleteData
+}
+
+func (s *service) allCommands() map[string]commandHandler {
+	uninstallAC := model.NewAutocompleteData("uninstall", "", "Uninstall an app")
+	uninstallAC.AddTextArgument("ID of the app to uninstall", "appID", "")
 
-		subTrigger = append(subTrigger, t)
+	all := map[string]commandHandler{
+		"info": {
+			f:            s.executeInfo,
+			autoComplete: model.NewAutocompleteData("info", "", "Display debugging information"),
+		},
+		"list": {
+			f:            s.executeList,
+			autoComplete: model.NewAutocompleteData("list", "", "List installed and registered apps"),
+		},
+		"uninstall": {
+			f:            s.checkSystemAdmin(s.executeUninstall),
+			autoComplete: uninstallAC,
+		},
 	}
 
-	sort.Strings(subTrigger)
+	conf := s.conf.GetConfig()
+
+	if conf.DeveloperMode {
+		debugAddManifestAC := model.NewAutocompleteData("debug-add-manifest", "", "Add a manifest to the local list of known apps")
+		debugAddManifestAC.AddNamedTextArgument("url", "URL of the manifest to add", "URL", "", true)
 
-	helpText := "Available commands: "
-	for i, t := range subTrigger {
-		if i == 0 {
-			helpText += t
-		} else {
-			helpText += ", " + t
+		all["debug-bindings"] = commandHandler{
+			f:            s.executeDebugBindings,
+			devOnly:      true,
+			autoComplete: model.NewAutocompleteData("debug-bindings", "", "List bindings"),
+		}
+		all["debug-clean"] = commandHandler{
+			f:            s.executeDebugClean,
+			devOnly:      true,
+			autoComplete: model.NewAutocompleteData("debug-clean", "", "Delete all KV data"),
+		}
+		// TODO ticket: change to watch-manifest
+		all["debug-add-manifest"] = commandHandler{
+			f:            s.executeDebugAddManifest,
```

Comment:
Given that is coding is getting touched: Should `s.checkSystemAdmin` be added here for consistency?